### PR TITLE
SPARKTA-362 Refactor in GeoHashField

### PR DESCRIPTION
**Pull request by dvallejo**
Refactor of the GeoHashField class.

-> [Ticket in JIRA](https://stratio.atlassian.net/browse/SPARKTA-362)

### CODE REVIEW

**Comment by compae** [comment]
+1, but when merge my PR you must change all JSerializable on the methods to Any. But not in the Map configuration object

**Comment by dvallejo** [comment]
@compae I'll do it. Thanks!

File: plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala [hunk 1]
```diff
@@ -81,69 +79,65 @@ case class GeoHashField(props: Map[String, JSerializable])
...
-      } else {
-        log.info("The geolocation precision can not be casted to Option")
-        (defaultPrecision, getPrecision(0, 0, defaultPrecision))
+  def precisionValue(keyName: String, value: JSerializable): (Precision, JSerializable) =
+    Try {
+      Option(value) match {
+        case Some(latLongValue: String) => extractPrecision(keyName, latLongValue)
+        case None => defaultPrecisionTuple
+        case _ =>
+          log.info("The geolocation precision can not be casted to Option")
```
Comment (danielcsant) [inline plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala]: Shouldn't be this log.error?

Comment (dvallejo) [inline plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala]: Yes, I'm going to change it.

Comment (dvallejo) [inline plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala]: Done

**Pull request closed without merging**